system_modules/naposc/src/oscsender.cpp

```diff
@@ -31,22 +32,38 @@ namespace nap
 		char hostIpAddress[IpEndpointName::ADDRESS_STRING_LENGTH];
 		host.AddressAsString(hostIpAddress);
 
-		// Create socket
-		mSocket = std::make_unique<UdpTransmitSocket>(host);
-		nap::Logger::info("Started OSC output connection, ip: %s, port: %d", hostIpAddress, mPort);
+		// Try to create the socket
+        mSocket = nullptr;
+        try
+        {
+            mSocket = std::make_unique<UdpTransmitSocket>(host);
+            nap::Logger::info("Started OSC output connection, ip: %s, port: %d", hostIpAddress, mPort);
+        }catch (const std::exception& e)
+        {
+            // Exception can be thrown in constructor of UdpTransmitSocket, caught here
+            // If we allow failure, log the error, otherwise return false and fail initialization
+            if(!errorState.check(mAllowFailure, "Failed to start OSC output connection, ip: %s, port: %d, error: %s", hostIpAddress, mPort, e.what()))
+                return false;
+
+            nap::Logger::error("Failed to start OSC output connection, ip: %s, port: %d, error: %s", hostIpAddress, mPort, e.what());
+        }
 
 		return true;
 	}
 
 
 	void OSCSender::stop()
 	{
-		mSocket.reset(nullptr);
-	}
+        if(mSocket!=nullptr)
+            mSocket.reset(nullptr);
+    }
 
 
 	bool OSCSender::send(const OSCEvent& oscEvent)
 	{
+        if(!mSocket)
```

Comment:
We encourage the use of an explicit `nullptr` check, instead of the implicit one used here. This tells the user the socket is a pointer and ensures that (in the unfortunate) case the type is changed compilation will fail instead of siltently pass by implicitly casting to a boolean.